### ISSUE TITLE
Can I have multiple cdylib output artifacts in a single project?

### ISSUE THREAD

**Issue by pepyakin** [rustwasm/team#95]
I've tried to set up my `Cargo.toml` as:

```toml
[[lib]]
name = "foo"
path = "src/foo.rs"

[[lib]]
name = "bar"
path = "src/bar.rs"
```

but Cargo complains

```
invalid type: map, expected a string for key `lib`
```

I know such mechanism works fine for `[[bin]]`, but for wasm32-unknown-unknown I guess I need `lib`.

**Comment by badboy** [comment]
Cargo does not have support for multi-lib projects. See [an old thread for a discussion](https://internals.rust-lang.org/t/how-about-changing-lib-to-lib-to-allow-multiple-library-in-a-crate/2022/3)
By now we have cargo workspaces though, which allow to have multiple libraries under a single main workspace and still act independent.

**Comment by pepyakin** [comment]
Yeah, I'm thinking about solving the issue with parity's wasm contract test repo.

At the Emscripten epoch (or era? :D) we just enumerated every test in our Cargo.toml , like [that](https://github.com/paritytech/wasm-tests/blob/28fc1e21f5712ed7f59e081330cc4c56e6405646/Cargo.toml). It worked fine, since there is a lot of small binaries.

However, with migration to wasm32-unknown-unknown (thus multi-lib project) we started [to generate](https://github.com/paritytech/wasm-tests/blob/master/build-test.sh) a project-dir for each test, which looks very ugly.
